### ISSUE TITLE
Stuck on Loading Music Library

### ISSUE THREAD

**Issue by hoghug** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#3660]
**OS:** macOS Mojave 10.14.5

**GPMDP Version:** 4.6.1

**Issue Descriptions:**
Stuck on Loading Music Library after opening

<img width="1388" alt="Screen Shot 2019-10-31 at 12 22 09 PM" src="https://user-images.githubusercontent.com/6405882/67979308-38be9580-fbd9-11e9-8a6c-c39a9965f231.png">

**Comment by Aer0nic** [comment]
I'm having the same issue, Win 10. Had it happen on multiple systems. Youtube Music works just fine though.

**Comment by MarshallOfSound** [comment]
Fixed in master, will hopefully release soon.  You can download builds from CI in the meantime

**Comment by nintyuk** [comment]
I don't understand, what is the fix? I've uninstalled and installed the latest version but still can't get it to load my library

**Comment by jfries289** [comment]
> Fixed in master, will hopefully release soon. You can download builds from CI in the meantime
@MarshallOfSound 
TravisCL for OS X appears to be inactive and 2 years out of date. Can we get an OS X build?

**Comment by MarshallOfSound** [comment]
@jfries289 The OSX and Linux builds are on CircleCI.

**Comment by Gibbel91** [comment]
I have the same issue on Windows 10.

**Comment by jfries289** [comment]
@Gibbel91 You can find the latest windows build to download at https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/build/artifacts

I believe it has the fix for this in it.

**Comment by edorgeville** [comment]
Thanks @jfries289, this fixes this issue for me (Windows 10).

**Comment by singjsong** [comment]
For lazy OS X users: https://3667-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip
May become dead in the future, I don't know how long Circle retains artifacts.

Thanks a lot for the quick fix, btw

**Comment by Nephiel** [comment]
Stumbled upon this today. For Debian amd64: https://3667-40008106-gh-circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.6.1_amd64.deb

**Comment by jdobbin-hsahealthplan** [comment]
The link @Nephiel gave didn't work for me, but this one did:

https://3667-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.6.1_amd64.deb

**Comment by Bobyn** [comment]
> The link @Nephiel gave didn't work for me, but this one did:
> 
> https://3667-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.6.1_amd64.deb

Thank you, It worked for me.

**Comment by Tsumiki** [comment]
> 
> 
> @Gibbel91 You can find the latest windows build to download at https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/build/artifacts
> 
> I believe it has the fix for this in it.

Thank you! Getting latest CI build fixed for me too :).

**Comment by jayDB5** [comment]
> @Gibbel91 You can find the latest windows build to download at https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/build/artifacts
> 
> I believe it has the fix for this in it.

Don't exactly know what this means, but it fixed the issue for me too - thanks!

**Comment by ctron** [comment]
Is there any chance to get a  4.6.2 version with this fix?

**Comment by cdyax** [comment]
I can connect perfectly with the version for youtube but not for google play, it gets to say that songs load and stay there.
I'm in Spain.
version 4.6.1
Google the web of play I have no problem with my email and password
If anyone has any idea how can I fix it agradeceria.

**Comment by bigclap** [comment]
Same on ubuntu 19.10

**Comment by cdyax** [comment]
> 
> 
> > @Gibbel91 You can find the latest windows build to download at https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/build/artifacts
> > I believe it has the fix for this in it.
> 
> Don't exactly know what this means, but it fixed the issue for me too - thanks!

**Comment by austinbutler** [comment]
Finding where to download the latest MacOS build was confusing to me, but here's a working link for all but Windows: https://circleci.com/api/v1.1/project/github/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/latest/artifacts

**Comment by dcikpeama** [comment]
I seem to have now run into the same issue on my PC as well. I've tried reinstalling the latest version, but as soon as I open up GPMDP, it just stays on "Loading Music Library..."

**Comment by doubleloop** [comment]
@MarshallOfSound 

>will hopefully release soon

Is there anything blocking the release of current master as is?

**Comment by iamhenry** [comment]
I just came across this thread and I'm having the same issue on a Mac. Does anyone know what I need to do in order to resolve this?

**Comment by rileym94** [comment]
Just wanted to throw out a note to say that I'm running Windows 10 1809 x64, and I very suddenly started having the player hang on the "loading music library" this afternoon following a reboot. Tried uninstall/reinstall, clearing %appdata% and %localappdata% files and reinstalling, and was spinning my wheels till I found this pinned. Thanks to @jostrander for pinning it. 
If you were too rushed to read the whole issue... Download a build from CI and keep an eye out for the next master release. 
[https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/build/artifacts](https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/build/artifacts)

**Comment by rileym94** [comment]
> I just came across this thread and I'm having the same issue on a Mac. Does anyone know what I need to do in order to resolve this?

@iamhenry I haven't seen any CI links for macOS builds, only Windows and Linux. Maybe @jostrander or @MarshallOfSound can point you in the right direction?

**Comment by daiyi** [comment]
The Mac builds are on CircleCI under the name "Darwin". You can go to a successful build, such as [this page](https://app.circleci.com/jobs/github/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/3727/artifacts), go to the Artifacts tab, and click `home/circleci/project/dist/installers/darwin/Google Play Music Desktop Player.zip` to download. 

After attempting to open the app, I had to go to Security & Privacy settings and make an exception to allow the mysterious internet-downloaded app to launch.

**Comment by iamhenry** [comment]
> The Mac builds are on CircleCI under the name "Darwin". You can go to a successful build, such as [this page](https://app.circleci.com/jobs/github/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/3727/artifacts), go to the Artifacts tab, and click `home/circleci/project/dist/installers/darwin/Google Play Music Desktop Player.zip` to download.
> 
> After attempting to open the app, I had to go to Security & Privacy settings and make an exception to allow the mysterious internet-downloaded app to launch.

@daiyi  I'm getting "Page Not Found" with that link you provided to the artifacts page. Can you provide an alternative working link plz?

![image](https://user-images.githubusercontent.com/923439/70281630-62d31c80-1770-11ea-8670-bf459b43d98c.png)

**Comment by DASPRiD** [comment]
For me this issue just popped up today also on stable 4.6.1 version. Is there a workaround for this issue?

**Comment by yioteh** [comment]
> For me this issue just popped up today also on stable 4.6.1 version. Is there a workaround for this issue?

Same. Happened to me on macos mojave 10.14.6 just today
Fixed by:
1. going to https://circleci.com/api/v1.1/project/github/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/latest/artifacts and downloading file with link having `darwin` portion in it - for macos. There're also debian (ubuntu) and redhat versions. Working windows version is located [here](https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/build/artifacts) 
2. removing existing installed app 
3. installing downloaded file
4. ...
5. PROFIT!

**Comment by Slapbox** [comment]
@yioteh - thank you! Quality of life restored! And of course thanks to all the creators of this great software!

**Comment by DASPRiD** [comment]
@yioteh Sadly, in that version my media hot keys do not work.

**Comment by titomb345** [comment]
Experiencing this issue on both Win7 and macOS 10.14.6

**Comment by Zoodinger** [comment]
Experiencing this issue with Windows 10 even though I'm using latest version.

**Comment by KicknGuitar** [comment]
I'm chiming in to say this issue came up on version 4.6.1 for both Windows and Mac but the link provided above from singjsong solved the issue.

**Comment by renclav** [comment]
(mac version)
This worked for me:
https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3660#issuecomment-562656210

But I'd like to point out that switching to the youtube player was a dead end, as it resulted in a "not available for your browser" page, with the usual burger menu option to switch back missing.
The fix:
editing the settings.json file with the following value:
"service": "google-play-music"

**Comment by Sirroga** [comment]
Fixed version solved the main problem but now media hot keys stopped working.

**Comment by DASPRiD** [comment]
> Fixed version solved the main problem but now media hot keys stopped working (Win 10).

Had the same problem. Don't know why, but after a reboot the media hotkeys also worked on the new version.

**Comment by thecoloryes** [comment]
> Fixed version solved the main problem but now media hot keys stopped working (Win 10).

I have this same issue on Win 10. If I disable the "Show track information in Windows 10 volume overlay" option in the desktop setting, my multimedia key works as expected. Note that this setting only appears if you have the Windows 10 System Media Service option enabled. You can disable the entire service for the multimedia keys to work, but I only had issues when I enabled the "Show track information in Windows 10 volume overlay" option 

When it's enabled, the overlay will display "No track playing" even when there is music playing. If I use my multimedia key for play/pause, the overlay freaks out a little and toggles into some kind of play/pause loop where the overlay thinks it's playing but there's no audio coming out. I have to go back into the app to get the music to resume properly (and if I hit multimedia play/pause before actually playing any music the app totally freaks out and won't start any music).

If I can provide any more diagnostic data let me know!

**Comment by electrikweb** [comment]
I'm new here...When I visit https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/build/artifacts to download the latest Windows build, I see a message "The build job does not contain any artifacts." and can't see anything I can download. Am I missing something? Or am I not looking in the right spot?

**Comment by IndefiniteBen** [comment]
You need to click on [history](https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/history) and then find the most recent [successful build](https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/builds/29394771/artifacts) I guess.

**Comment by ruborg** [comment]
I've been using this on Ubuntu for > 1 year, and it has been working great until today. Then out of the blue, it's stuck on 'Loading Library Music Library'.

I restarted machine, downloaded and installed the latest .deb, and still same problem. This is the output I get:

```
rods@rods-XPS:~$ google-play-music-desktop-player 
Starting Sentry
error: Failed to load bonjour with error: {}
Bonjour is required to use Chromecast Support or to enable ZeroConf for the PlaybackAPI
On linux you need to install "avahi"
Failed to encode dictionary
Failed to encode reply value
Failed to encode dictionary
Failed to encode reply value
```

**Comment by ruborg** [comment]
Reinstalling as per this page:

`https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/blob/master/docs/Installation_On_Debian.md`

Ends up outputting these errors:

```
error: Emitter fn() threw exception. TypeError: Cannot read property '0' of undefined
    at LibraryNamespace.getTracks (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/gmusic-ui.js/dist/gmusic-ui.js:107:84)
    at LibraryNamespace.getLibrary (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/gmusic-ui.js/dist/gmusic-ui.js:99:22)
    at /usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/playback/handler.js:100:53
    at /usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/index.js:81:9
    at arrayEach (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/lodash/lodash.js:537:11)
    at Function.forEach (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/lodash/lodash.js:9359:14)
    at /usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/index.js:79:22
error: Emitter fn() threw exception. TypeError: Cannot read property 'insertBefore' of undefined
    at installSidebarButton (/usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/interface/customUI.js:121:58)
    at installDesktopSettingsButton (/usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/interface/customUI.js:127:3)
    at installMainMenu (/usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/interface/customUI.js:161:3)
    at /usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/interface/customUI.js:274:3
    at /usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/index.js:81:9
    at arrayEach (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/lodash/lodash.js:537:11)
    at Function.forEach (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/lodash/lodash.js:9359:14)
    at /usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/index.js:79:22
error: Emitter fn() threw exception. TypeError: Cannot read property '0' of undefined
    at LibraryNamespace.getTracks (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/gmusic-ui.js/dist/gmusic-ui.js:107:84)
    at LibraryNamespace.getLibrary (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/gmusic-ui.js/dist/gmusic-ui.js:99:22)
    at /usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/playback/handler.js:100:53
    at /usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/index.js:81:9
    at arrayEach (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/lodash/lodash.js:537:11)
    at Function.forEach (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/lodash/lodash.js:9359:14)
    at /usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/index.js:79:22
error: Emitter fn() threw exception. TypeError: Cannot read property 'insertBefore' of undefined
    at installSidebarButton (/usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/interface/customUI.js:121:58)
    at installDesktopSettingsButton (/usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/interface/customUI.js:127:3)
    at installMainMenu (/usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/interface/customUI.js:161:3)
    at /usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/interface/customUI.js:274:3
    at /usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/index.js:81:9
    at arrayEach (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/lodash/lodash.js:537:11)
    at Function.forEach (/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/lodash/lodash.js:9359:14)
    at /usr/share/google-play-music-desktop-player/resources/app.asar/build/renderer/windows/GPMWebView/index.js:79:22
    ```

**Comment by jostrander** [comment]
@ruborg, I might suggest using the snap --edge channel. The deb packages from that repo are extremely out of date.

**Comment by ruborg** [comment]
This worked, thanks for the tip!

**Comment by dkudriavtsev** [comment]
question: when I click on the artifacts page for successful builds, I only see windows .exe installers - where do I find the Linux builds?

**Comment by jostrander** [comment]
You are looking in AppVeyer. The linux builds are under the collect artifacts job of CircleCI.

**Comment by abion136** [comment]
Windows workaround: https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3640#issuecomment-543863416

Fixed the issue on my Win10 machine if the CI builds fail for anyone.

**Comment by electrikweb** [comment]
> 
> 
> You need to click on [history](https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/history) and then find the most recent [successful build](https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/builds/29394771/artifacts) I guess.

Perfect! Thank you. Now working again on my Windows 10 pro.

**Comment by jschewebbn** [comment]
I'm trying to find the Linux builds from the Circle CI link https://circleci.com/gh/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL- and it's not clear to me where to find the build outputs.

**Comment by alexbaucom** [comment]
Just had this issue happen to me this morning too. On Ubuntu 18.04 I was able to resolve this by uninstalling and then reinstalling from snap with the edge channel as mentioned above.
`sudo snap install google-play-music-desktop-player --edge`

**Comment by nzchicken** [comment]
For linux/mac users that want the latest build, easiest way is through here (rather than looking through circleci apis): https://circleci.com/gh/MarshallOfSound/workflows/Google-Play-Music-Desktop-Player-UNOFFICIAL-

1. Find the latest Green workflow
2. Click the artifact-gathering box
3. Click artifacts tab
4. Expand tree until you get to your installer. Darwin = Mac for those that don't know.

**Comment by AO0213** [comment]
This issue is still on going and not fixed.

**Comment by IndefiniteBen** [comment]
@AO0213 can you clarify which version you're using? If you are not using a version from 
AppVeyor/CircleCI/self-built, then you're still running the old version without the fix, as [mentioned above by MarshallOfSound](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3660#issuecomment-548558168):  
> Fixed in master, will hopefully release soon. You can download builds from CI in the meantime

**Comment by nzchicken** [comment]
On Mac, after using the latest version from circleci above, click "Try Youtube Music Player Desktop" beta button says: "Sorry Youtube Music is not optimised for our browser. Check for updates or try Google Chrome". I'm happy to build a version and play around with user agents if that helps?

**Comment by kp37** [comment]
Has anyone found a fix for this on Mac OS? I am running Catalina 10.15.1. I have tried to grab the new release but Mac Security won't let me install it.

**Comment by IndefiniteBen** [comment]
> Edit: Also, no idea how to shift back to GPMDP from YMPD either.

Mentioned by @jostrander in another issue thread, for both the issue with "not optimised" message and just how to switch back to GPM: right click on the icon in the tray and the top menu option is for choosing the service. You don't switch between services from inside the app window itself.

**Comment by AO0213** [comment]
> @AO0213 can you clarify which version you're using? If you are not using a version from
> AppVeyor/CircleCI/self-built, then you're still running the old version without the fix, as [mentioned above by MarshallOfSound](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3660#issuecomment-548558168):
> 
> > Fixed in master, will hopefully release soon. You can download builds from CI in the meantime

Version 4.6.1 on WIndows 10

**Comment by no452** [comment]
Version 4.6.1, Ubuntu. Also stuck and can't seem to get it working

**Comment by erosen03** [comment]
Just started having the library load issue on Windows 10 Pro. Using version 4.6.1. Was working fine as recently as yesterday.

**Comment by andrewsnydes** [comment]
Can confirm using GPMDP on Windows10 x64 version 4.6.1 stuck on loading screen across multiple devices.

**Comment by DeclanEM** [comment]
Confirm same issue started today.

**Comment by DummyHorse** [comment]
I have the app on multiple work computers and my own personal computer and all have the same issue. Started today, was fine before.

**Comment by graywizardx** [comment]
Seeing this with the snap package on Ubuntu 18.04: 
Error: dns service error: unknown
    at new Browser (/snap/google-play-music-desktop-player/72/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/mdns/lib/browser.js:86:10)
    at Object.create [as createBrowser] (/snap/google-play-music-desktop-player/72/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/mdns/lib/browser.js:114:10)
    at new _class (/snap/google-play-music-desktop-player/72/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/node-mdns-easy/dist/browser.js:44:30)
    at _class.createBrowser (/snap/google-play-music-desktop-player/72/usr/share/google-play-music-desktop-player/resources/app.asar/node_modules/node-mdns-easy/dist/mdns.js:42:14)
    at Object.<anonymous> (/snap/google-play-music-desktop-player/72/usr/share/google-play-music-deskt…resources/app.asar/node_modules/electron-chromecast/build/cast/index.js:60:20)
    at Object.<anonymous> (/snap/google-play-music-desktop-player/72/usr/share/google-play-music-deskt…esources/app.asar/node_modules/electron-chromecast/build/cast/index.js:275:3)
    at Module._compile (module.js:642:30)
    at Object.Module._extensions..js (module.js:653:10)
    at Module.load (module.js:561:32)
    at tryModuleLoad (module.js:504:12)

And this in the console:
*** WARNING *** The program 'Google Play Music Desktop Player' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
*** WARNING *** The program 'Google Play Music Desktop Player' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
debug: Renderer process logger initialized.
*** WARNING *** The program 'Google Play Music Desktop Player' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
*** WARNING *** The program 'Google Play Music Desktop Player' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
debug: Renderer process logger initialized.
*** WARNING *** The program 'Google Play Music Desktop Player' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
*** WARNING *** The program 'Google Play Music Desktop Player' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>

According to Snap:
snap info google-play-music-desktop-player
name:      google-play-music-desktop-player
summary:   Standalone player for Google Play Music
publisher: Chris Halse Rogers (cooperteam)
contact:   https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues
license:   unset
description: |
  This Google Play Music player provides desktop integration - with music
  hotkeys and the sound indicator via an MPRIS interface, the abiliy to use
  custom colours and load custom themes, a remote-control interface, and
  lower resource consumption than a full-blown browser tab.
  .
  This is *not* an official Google project.
commands:
  - google-play-music-desktop-player
snap-id:      4Y41wz5UioPArGBhEyQaO1DkY5pfUEpa
tracking:     stable
refresh-date: today at 19:34 PST
channels:
  stable:    4.6.1              2019-09-13 (72) 116MB -
  candidate: 4.6.1              2019-09-13 (72) 116MB -
  beta:      4.6.1              2019-09-13 (72) 116MB -
  edge:      v4.6.1-63-g4445ffa 2019-12-09 (77)  98MB -
installed:   4.6.1                         (72) 116MB -

**Comment by abion136** [comment]
@nzchicken check out this https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3545#issuecomment-563215650 which fixed YTM for me. Just follow the instructions per the linked comment.

**Comment by abion136** [comment]
@ruborg have you tried to build it yourself from the master branch? I would suggest either doing that or getting the latest CI build to test.

**Comment by WebFactoryLtd** [comment]
Can confirm too, on Win10. Since yesterday.

**Comment by buzzik** [comment]
Same here

**Comment by slipperydippery** [comment]
Ran into this issue today, Win10

**Comment by shoaib30** [comment]
facing the same issue on MacOS.

**Comment by carolfs** [comment]
I ran into this problem on MacOS, but installing this version fixed it: https://3667-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip

**Comment by troylit** [comment]
Confirmed, same issue on Windows 10 64 bit v1909 Build 18363.535 running GPMDP v 4.6.1.

**Comment by jarayama** [comment]
The mac version not even loads the window, just hangs after opening the app.

**Comment by Hank3** [comment]
Add me to the list.  Google Play Music Desktop was working as of 12.10.19 @ 5 PM ET.  This it was stuck on the "Loading Music Player" screen.  I'm running Windows 10 Pro 64-bit.

**Comment by bruno-brant** [comment]
I'm having the same issue, Win10. The app was working up until last Friday. YT works.

**Comment by eKalb33** [comment]
Stopped working mid afternoon yesterday for me as well. Win 10.

**Comment by bruno-brant** [comment]
I just downloaded a build from appveyor and worked for me.

https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/builds/29394771/artifacts

Not sure how stable those builds are.

**Comment by juanmagalhaes** [comment]
I'm having this issue on Kubuntu 18. I'm on the latest release:

https://3667-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.6.1_amd64.deb

I also tried downgrading to the previous release and blowing up the local config under `.config/Google Play Music Desktop Player` but nothing seemed to work so far :disappointed:

**Comment by Teascade** [comment]
> 
> 
> @Gibbel91 You can find the latest windows build to download at https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/build/artifacts
> 
> I believe it has the fix for this in it.
How do I download anything from this website, exactly?

**Comment by jostrander** [comment]
https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/builds/29394771/artifacts

This is the latest master build. The link you posted was the latest build in general, which was from a PR and it failed, which is why there are no artifacts.

**Comment by Hank3** [comment]
> https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/builds/29394771/artifacts
> 
> This is the latest master build. The link you posted was the latest build in general, which was from a PR and it failed, which is why there are no artifacts.

I downloaded the version you posted and when I run/open GPMD, I get a YouTube Music error:

Sorry, YouTube Music is not optimized for your browser. Check for updates or try Google Chrome.

I already have Chrome, but if I download it within the GPMD window it stays in the Chrome window.

**Comment by jostrander** [comment]
Unfortunately this and another issue (#3545) have both popped up at the same time, so until #3545 is closed the master builds won't be 100% working for everyone.

**Comment by Hank3** [comment]
Finally got things working again by downloading the latest master build posted; closing GPMDP and deleting the cache logs.

**Comment by slipperydippery** [comment]
@Hank3 how did you delete the cache logs?

**Comment by Hank3** [comment]
> @Hank3 how did you delete the cache logs?

I used the info from this post:  https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3707#issuecomment-562957745

**Comment by slipperydippery** [comment]
Thanks for the reply, however now it won't let me log in again 

https://support.google.com/accounts/answer/7675428

**Comment by andrewsnydes** [comment]
Doing this allowed YTMusic to start and get into library but still can't get into GPM. (was using taskbar right click menu to switch)
I can get into desktop settings also but thats it. I've had this issue for about a full week now.

**Comment by andrewsnydes** [comment]
Using this build though does in fact fix this problem:
https://ci.appveyor.com/api/buildjobs/amjhy1vx6gjpf31k/artifacts/dist%2Finstallers%2Fwin32%2FGoogle%20Play%20Music%20Desktop%20PlayerSetup.exe

**Comment by tauil** [comment]
I'm having the same issue (stuck Loading like on the screen at the beginning of the issue). I already deleted the whole Google Play Music App folder in the Application Support folder (Mac) and it didn't work. I tried on two different computers.

**Comment by ollien** [comment]
> https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/builds/29394771/artifacts
> 
> This is the latest master build. The link you posted was the latest build in general, which was from a PR and it failed, which is why there are no artifacts.

Is there a way to get this build for Linux (RPM distros, specifically)? The linked build only has a Windows installer.

**Comment by jostrander** [comment]
https://3751-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.6.1.x86_64.rpm

**Comment by ollien** [comment]
Ah, I didn't realize there were two different build systems at work, here. Thanks!

**Comment by NovaViper** [comment]
Im having the same issue here, it just started today. I have the Windows build

**Comment by abion136** [comment]
@NovaViper did you try the build that's linked 5 comments above yours?

**Comment by abion136** [comment]
@tauil have you tried this build: https://3667-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip?

**Comment by NovaViper** [comment]
> @NovaViper did you try the build that's linked 5 comments above yours?
> 
> https://ci.appveyor.com/api/buildjobs/amjhy1vx6gjpf31k/artifacts/dist%2Finstallers%2Fwin32%2FGoogle%20Play%20Music%20Desktop%20PlayerSetup.exe

That worked! Thanks

**Comment by bsurd** [comment]
I've also installed the build from [here](https://ci.appveyor.com/api/buildjobs/amjhy1vx6gjpf31k/artifacts/dist%2Finstallers%2Fwin32%2FGoogle%20Play%20Music%20Desktop%20PlayerSetup.exe). I can confirm that the issue is indeed fixed on Windows 10 in my case. 

Thanks for fixing this! I really love the desktop player and depend on it on a day to day basis 😄

**Comment by kp37** [comment]
> I ran into this problem on MacOS, but installing this version fixed it: https://3667-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip

How do you get around this?
![image](https://user-images.githubusercontent.com/21187665/70719758-94903a00-1cc0-11ea-9c03-521c4b561891.png)

**Comment by Pobega** [comment]
Any idea when we'll get a release with this fix? Currently the [Linux Flatpak](https://github.com/flathub/com.googleplaymusicdesktopplayer.GPMDP) is suffering from this issue but they track releases, not master.

**Comment by samiconductor** [comment]
I'm in the same boat as @Pobega. What can we do to help get a release with this fix out there?

**Comment by jottr** [comment]
> > I ran into this problem on MacOS, but installing this version fixed it: [3667-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip](https://3667-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip)
> 
> How do you get around this?
> ![image](https://user-images.githubusercontent.com/21187665/70719758-94903a00-1cc0-11ea-9c03-521c4b561891.png)

@kp37 
This is unrelated to this issue. Googling should have helped you figure this out yourself easily. You can start [here](https://support.apple.com/en-us/HT202491).

**Comment by jottr** [comment]
To save others from having to hunt down the most recent MacOS builds.

1. Identify the most successful build for `Darwin` on [CircleCI](https://circleci.com/gh/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/tree/master).  
2. Just plug the most recent build number into the following oneliner: 

```
BUILD=3729 wget "$(curl -s https://circleci.com/api/v1.1/project/github/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/3729/artifacts | grep -o 'https://[^"]*')"
```

Build `3729` works for me.

**Comment by agzam** [comment]
Can someone tell me why can't we have a brew formula for this app? So we can do `brew install gpmdp --HEAD`, instead every time having to go through hurdles whenever Google breaks API on a whim?

**Comment by jostrander** [comment]
@agzam It's open source, feel free to submit a PR or write one. :+1:

**Comment by agzam** [comment]
Oh, there is a formula, the name of it makes it difficult to find.
`brew cask info marshallofsound-google-play-music-player`
sadly, it does not seem to support --head flag

**Comment by c4naan** [comment]
> @NovaViper did you try the build that's linked 5 comments above yours?
> 
> https://ci.appveyor.com/api/buildjobs/amjhy1vx6gjpf31k/artifacts/dist%2Finstallers%2Fwin32%2FGoogle%20Play%20Music%20Desktop%20PlayerSetup.exe

this also worked for me. thank you

**Comment by CindyBlendy** [comment]
This link no longer has a build for Windows 10 users: https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/build/artifacts

**Comment by svpernova09** [comment]
(MacOS Catalina)

This morning I ran into this issue, and before searching I just instinctively wiped out `~/Library/Application\ Support/Google Play Music Desktop Player` and then couldn't log back in. I was getting "Browser not secure, could no log you in" errors. 

As of this comment this is the latest Darwin build: `BUILD=3744 wget "$(curl -s https://circleci.com/api/v1.1/project/github/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/3729/artifacts | grep -o 'https://[^"]*')"`

Thanks to @jottr for the comment https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3660#issuecomment-565094820

**Comment by tjkirch** [comment]
Would definitely like to see a release so we can get a flatpak update.  I haven't been able to use any other Linux version of GPMDP  for years because it just gives a white screen (which other opened issues have not resolved) so I have to use the flatpak.

**Comment by dmitriy461** [comment]
> @NovaViper did you try the build that's linked 5 comments above yours?
> 
> https://ci.appveyor.com/api/buildjobs/amjhy1vx6gjpf31k/artifacts/dist%2Finstallers%2Fwin32%2FGoogle%20Play%20Music%20Desktop%20PlayerSetup.exe

Whoa, this did it! The regular build kept giving trouble this morning (after rebooting for Windows updates), but this one solved it.

**Comment by jwhipp** [comment]
Is the plan to use this appveyor now instead of the GH releases section?  It just barely noticed that's from over 1 year ago.  O.o

**Comment by tjkirch** [comment]
For those using the flatpak app, I found a workaround until a release is cut and flathub is updated.  Basically, rebuild the flatpak and point it at the new .deb listed above.

* Get the flatpak source: `git clone 'https://github.com/flathub/com.googleplaymusicdesktopplayer.GPMDP.git'` and cd in
* Install dependencies:
  * `flatpak install flathub org.freedesktop.Sdk/x86_64/18.08`
  * `flatpak install flathub io.atom.electron.BaseApp//18.08`
* Edit `com.googleplaymusicdesktopplayer.GPMDP.json`:
  * Change the `url` key from github to the CI artifact link above, `https://3667-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.6.1_amd64.deb`
  * Change the `sha256` just below it to the sha256 of the file
* Build: `flatpak-builder build-dir com.googleplaymusicdesktopplayer.GPMDP.json`
* Now you can run using: `flatpak-builder --run build-dir com.googleplaymusicdesktopplayer.GPMDP.json google-play-music-desktop-player`

**Comment by dmitriy461** [comment]
> > @NovaViper did you try the build that's linked 5 comments above yours?
> > https://ci.appveyor.com/api/buildjobs/amjhy1vx6gjpf31k/artifacts/dist%2Finstallers%2Fwin32%2FGoogle%20Play%20Music%20Desktop%20PlayerSetup.exe
> 
> Whoa, this did it! The regular build kept giving trouble this morning (after rebooting for Windows updates), but this one solved it.

The one issue I'm stuck with is that for some reason the native Windows media controls work only half way -- I'm able to pause, but unable to resume. So I'm having to use the hotkeys, which is not as convenient. 

Not sure if that's an issue with my setup, or other people have the same issue?

**Comment by austin-xgen** [comment]
Still stuck on loading on ubuntu on https://3667-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.6.1_amd64.deb

**Comment by WebFactoryLtd** [comment]
> The one issue I'm stuck with is that for some reason the native Windows media controls work only half way -- I'm able to pause, but unable to resume. So I'm having to use the hotkeys, which is not as convenient.
> Not sure if that's an issue with my setup, or other people have the same issue?

I have exactly the same issue! And it also says "no track playing" in the Windows  Media Service / Volume overlay. Wasn't like that before :( Maybe a Windows update messed with it :(

**Comment by NovaViper** [comment]
When will an official patch for this be released??

**Comment by Stevester118** [comment]
Currently having this issue, stuck on the google play music loading screen

**Comment by robert9422** [comment]
I also have this issue, stuck on the google play music loading screen.
I have the latest version.

**Comment by robert9422** [comment]
Sorry, this link fixed it, for me. thanks.
https://ci.appveyor.com/api/buildjobs/amjhy1vx6gjpf31k/artifacts/dist%2Finstallers%2Fwin32%2FGoogle%20Play%20Music%20Desktop%20PlayerSetup.exe

**Comment by ciarpame** [comment]
I had the same issue, the build mentioned above fixed it but my Chromecast Audio is not detected anymore (it works as expected via https://play.google.com/ in a Chrome tab).

**Comment by Darel13712** [comment]
Happens again on Catalina

**Comment by yarko686** [comment]
Same issue here. MacOS Catalina 10.15.2, GPMDP Version 4.6.1

**Comment by agc93** [comment]
Are we going to see a 4.6.2 release?

It's been a year and a half since 4.6.1 and there's a ton of major/breaking problems for different platforms/distributions that have cropped up in that time.

Plus, some of the packages (Flatpak, Chocolatey) are now running a *long* way behind.

**Comment by droM4X** [comment]
Latest Darwin build (3744) working on MacOS Catalina 10.15.2
https://3744-40008106-gh.circle-artifacts.com/0/Users/distiller/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip

**Comment by yllekz** [comment]
Running into this issue on Windows right now.

**Comment by IndefiniteBen** [comment]
@jostrander Hesitant to tag MarshallOfSound... Is 4.7.0 on a short timetable before release? I haven't seen a **short** list that needs to be completed, so I guess it's not making release anytime soon. 
As mentioned by @agc93 a 4.6.2 release with some of the basic fixes like for this loading issue could reduce a lot of the recent issue reports? I don't know why it's happening lately (instead of June), but it seems to be affecting most users these days. 

Alternatively, maybe MarshallOfSound can edit his comment at the top of the thread with links to where to find the working builds?

> Are we going to see a 4.6.2 release?
> 
> It's been a year and a half since 4.6.1 and there's a ton of major/breaking problems for different platforms/distributions that have cropped up in that time.

**Comment by pete360** [comment]
Hello, Has this been fixed for Windows 10? I downloaded the latest version (4.6.1) on Windows 10 (Build 1909) and it's stuck on **Loading Music Library** at launch.

**Comment by jostrander** [comment]
@IndefiniteBen I don't make any decisions regarding releases, and I don't have the keys to sign anything for releases. I know that the outstanding login issue and user agent compatibility with YTM is a major blocker for doing a release currently. We can probably start gearing towards a release once that is finalized. Then we have to update the changelog, tag git, prepare releases through staging server for testing (All MarshallOfSound), and publish the release. I don't see there being a 4.6.2 unfortunately. If anything 4.7.0 will be the release version still, because there are 60+ commits in the backlog since 4.6.1, so it wouldn't be a small patch.

**Comment by IndefiniteBen** [comment]
@jostrander I didn't expect you made the decisions, but judging by your comment you definitely know enough to make it clear that any kind of release (like @agc93 suggested) **before** 4.7.0 is impractical and would be a waste of time at this point. 
In the meantime, for everyone who's still running into the issue, see the workaround (only personally tested on Windows). 

## Workaround 
1. Go to the build service website (history section):
   * Windows: [AppVeyor](https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/history)
   * Mac/Linux systems: [CircleCI](https://circleci.com/gh/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/tree/master) or [its workflows page](https://circleci.com/gh/MarshallOfSound/workflows/Google-Play-Music-Desktop-Player-UNOFFICIAL-)
2. Find the most recent (latest @20191216 linked) build that succeeded (which are the green ones).
   * Windows: https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/builds/29394771
   * Mac/Linux: https://circleci.com/gh/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/3751
3. Find the artifacts page and download the file:
    * [Windows](https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial/builds/29394771/artifacts)
   * Mac/Linux: apparently there should be an artifacts tab on the CircleCI/workflows links, but I can't find them so can't mention how right now... There's direct links to the most recent versions earlier in the thread, and also instructions on how to download and build it. Alternatively, see [this comment](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3660#issuecomment-565094820) for using `wget` to update it on Mac.
4. Install the new version and it should work! 

If anyone wants to tell me where I'm wrong so I can edit my comment I welcome that.

**Comment by jwhipp** [comment]
In terms of getting/running new versions, are the build instructions in the README.md up2date?  I haven't personally tried the npm builds and really haven't looked into the CI/CD configs, but if that works, it's another method.   Not that it's everyone's cup of tea, but another method nonetheless.

**Comment by MadRay** [comment]
Linux builds for build 3751 #master:

https://3751-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.6.1_amd64.deb
https://3751-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.6.1_i386.deb

https://3751-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.6.1.i386.rpm
https://3751-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.6.1.x86_64.rpm

**Comment by Darel13712** [comment]
@IndefiniteBen That command does not work for me. It says "the name of the server is unacceptable".

**Comment by Darel13712** [comment]
Ok. For anyone who can't make that command line work -- just download https://3751-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip
for mac. 

Just make sure to NOT click on button Youtube Music (Beta) It will brick app to a browser mode on YTM page and there's no way to go back. (Or is it? I donno, help me pls)

**Comment by droM4X** [comment]
> there's no way to go back. (Or is it? I donno, help me pls)

Just click on it's icon on menu bar and choose Google Play Music as service to switch back.

**Comment by Darel13712** [comment]
No such option

**Comment by droM4X** [comment]
![image](https://user-images.githubusercontent.com/10895544/70946079-d3264b80-2056-11ea-803c-f44273cfc203.png)

**Comment by GreatBigWhiteWorld** [comment]
Is there a way to find the link of downloads instead of clicking other's download link directly? No offense to people who's trying to help, but years of computer common sense won't let me just download from a direct link and install it that I never know where it's from.

**Comment by abion136** [comment]
@GreatBigWhiteWorld here's the CI links from the README:

https://ci.appveyor.com/project/MarshallOfSound/google-play-music-desktop-player-unofficial
https://circleci.com/gh/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-

**Comment by agc93** [comment]
@jostrander I understand you can't control it, but given that 4.7.0 is likely to be the next release to contain all these fixes (some of which are pretty fatal to the packaged versions*), is there any expectation, from what you can tell, that this release will come in any reasonable time frame? The release issue has been open for 7 months already...

---

While downloading CI artifacts can work around some of these bugs (as shitty a user experience as it is), forcing package maintainers to either a) field 18 months worth of bug reports waiting on an unresponsive upstream or b) maintain and triage packages based on CI-quality builds, seems like a pretty poor compromise.

**Comment by Darel13712** [comment]
@droM4X Many thanks! Sorry for stupidity.

Other question is there a way to remove this annoying YTM button? Or what release does not have it?

**Comment by GreatBigWhiteWorld** [comment]
@abion136 Thank you for the link. How come I can't find any link to a download from those pages?

**Comment by Darel13712** [comment]
@GreatBigWhiteWorld If you're not suspicious about oneliner posted [earlier](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3660#issuecomment-565094820) you can check link from that command -- https://circleci.com/api/v1.1/project/github/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/3729/artifacts

You can change the number of build to 3751.  And find the link I gave you in the darwin section of the response.

Don't know how can anyone retrieve this links from the circleci in a normal way

**Comment by GreatBigWhiteWorld** [comment]
@Darel13712 Thank you. Don't know why this has to be this unusually hard.

**Comment by derwent-m-js** [comment]
How feasible would it be to get someone who has access upload the latest version to the GitHub releases? It would save a lot of people a lot of googling and messing with CircleCI links :)

**Comment by abion136** [comment]
@GreatBigWhiteWorld the typical workflow would be to visit the CI page, find the most recent passing build for your platform, click its link, and then find the artifact section for that build. You want to download the artifact produced by the build - that is your .exe, .deb, etc.

Although, I can't seem to find a passing build on CircleCI that actually has an artifact section so... I'm building from source, so thankfully I don't actually have to deal with that issue though. I honestly don't know how people are finding artifacts to link to for Linux/OSX.

> Don't know why this has to be this unusually hard.

Normally, it isn't :stuck_out_tongue:

**Comment by FernandoMiguel** [comment]
I've opened a PR to update brew cask using a CI build until a new release is cut
https://github.com/Homebrew/homebrew-cask/pull/74407

**Comment by agc93** [comment]
Okay for anyone getting confused by the different links and CI providers, I've slapped together something to help you find the latest release...

[GPM Release Finder](https://gpmdp.azurewebsites.net) will just poll through the CI build history on AppVeyor or CircleCI and pull the last successful build on `master` for your chosen platform. It's hardly bulletproof but easier than keeping track of all the URLs being thrown around IMO.

[https://gpmdp.azurewebsites.net](https://gpmdp.azurewebsites.net)

**Comment by markgaze** [comment]
> Okay for anyone getting confused by the different links and CI providers, I've slapped together something to help you find the latest release...
> 
> [GPMDP Release Finder](https://gpmdp.azurewebsites.net) will just poll through the CI build history on AppVeyor or CircleCI and pull the last successful build on `master` for your chosen platform. It's hardly bulletproof but easier than keeping track of all the URLs being thrown around IMO.
> 
> https://gpmdp.azurewebsites.net

Life saver! (Well, you stopped me having to waste my life figuring this out for myself 😆)

Thanks! 👍

**Comment by Darel13712** [comment]
@agc93 Nice job! It would be great if it would list a couple of last releases, not just the last one.

**Comment by agc93** [comment]
@Darel13712 If you create an issue at [agc93/gpmdp-release](https://github.com/agc93/gpmdp-release) I'll see what I can do. Ideally this app won't be needed for that long 😆

**Comment by turdfergusonpdx** [comment]
> **OS:** macOS Mojave 10.14.5
> 
> **GPMDP Version:** 4.6.1
> 
> **Issue Descriptions:**
> Stuck on Loading Music Library after opening
> 
> <img alt="Screen Shot 2019-10-31 at 12 22 09 PM" width="1388" src="https://user-images.githubusercontent.com/6405882/67979308-38be9580-fbd9-11e9-8a6c-c39a9965f231.png">

I'm having this same problem. Newbie here, can someone stats steps to fix?

**Comment by turdfergusonpdx** [comment]
> Ok. For anyone who can't make that command line work -- just download https://3751-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/darwin/Google%20Play%20Music%20Desktop%20Player.zip
> for mac.
> 
> Just make sure to NOT click on button Youtube Music (Beta) It will brick app to a browser mode on YTM page and there's no way to go back. (Or is it? I donno, help me pls)

Never mind, I downloaded this build and it's working fine now.

**Comment by jnr2820** [comment]
Hi all,
I reqested @agc93 's neat search tool be added to the README. I guess we'll wait and see if it is accepted. 
https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/pull/3745

**Comment by IndefiniteBen** [comment]
@turdfergusonpdx read this thread. 
https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3660#issuecomment-567304602

**Comment by pete360** [comment]
Windows 10.

Well, I got a little further but now have a new issue. I installed Revo Uninstaller and did a complete uninstall and wipe of GPMD. Did a fresh install and it bring me to sign in the page, after signing I get a message from Google:

> Couldn't sign you in
> 
> This browser or app may not be secure. Learn more
> 
> Try using a different browser. If you’re already using a supported browser, you can refresh your screen and try again to sign in.

![image](https://user-images.githubusercontent.com/4324474/71420305-84724600-2642-11ea-89d0-fd83fab7f980.png)

**Comment by cconeill** [comment]
Why is this issue closed given that the bug still exists?

**Comment by jostrander** [comment]
Standard procedure is the close the issue when a bug has been merged into the code. The bug still exists in the version that was released, but does not exist in master, or future releases.

**Comment by for7raid** [comment]
@jostrander Good practice, realy, but over the month I can't listen to music, and there is not any roadmap to release next **working** version. 
The current released version **doesn't work at all**, this is a critical issue, and a standart procedure to make a patch release 14.6.2, and only then close the issue. Customers can't use master branch, folk don't know what "master" and "brach" mean, they just want to download app and listen to music.

I tried to use CI artifacts, but is crashed too.
@MarshallOfSound  Any news to this?

**Comment by MarshallOfSound** [comment]
@for7raid just gonna stop you right there, this project is open source, their are no "customers".  Everyone using this project does so for free, under the license in this repo.

We're working on finding a fix for all folks and at that point we will release a new version.  No one gets paid to work on this though so patience is key here if you can't figure out how CI works.

**Comment by for7raid** [comment]
@MarshallOfSound  Excuse me, I know that is open source and UNOFFICIAL, and I have to say "thank you for all thees years", but right now you are losing your unpaid, but anyway customers, who can't use app. I afraid it can be very expensive pay for this community.
While you created the great public app, you got the responsibility to all users and does not have ability to hide behind "no pay no service". 155 comments here, more over 10 clones and grows every day.
If you have workaround - please clearly provide it (links to download, instructions), if you have roadmap - publish it.
All this is very untransparenty for us, but you have closed the issue and don't support users.
I'm not angry about this issue,  I (and I'm not alone) confused how you answer to us.
I can't find any commits linked to this issue. Can you provide it?

**Comment by IndefiniteBen** [comment]
@for7raid This is an open source project; everyone working on this probably does it as a hobby next to an actual job/study. 

I can understand how it's annoying if you just download it to use, but you should've been aware that it's an open source project and what that means. If you want something reliable, use what is officially supported (by Google) and just use GPM in a browser. Chrome/Edge beta will even let you install a site as an app using PWA support.

**Comment by WebFactoryLtd** [comment]
@for7raid don't like it? Don't use it! As simple as that! Build a better app, fork this one find a better one ...
And no, Marshall has absolutely no responsibility towards anyone. Everything here is provided "as is".

**Comment by IndefiniteBen** [comment]
@for7raid See #3548 for the next release super ticket.

**Comment by Darel13712** [comment]
@WebFactoryLtd  don't be so nasty asking people to leave the community. Nativefier or web version is a more stable way to use GPM now than GPMDP today and it's not an achievement for this community to be proud of. Don't act like there is no problem which @for7raid is referring to even if you don't like his wording.

@MarshallOfSound great you're joined the conversation. I myself don't intend to blame anyone here. Master branch stopped working for me several monthes ago. I use ci build or web version or other services. But let me rephrase what @for7raid was trying to convey here in my own words.

Repository says that current build is passing. Yet a lot of people have been experiencing this issue 3660 for a long period of time now. Current build does not work and is not usable. If it's passing then some tests are missing. Please consider adding info that this package is broken atm and there is work concerning this to readme and site. This is wrong that people have to find this thread asking the same questions of how to download ci build over and over again. It's simple respect to let people know what's going on.

**Comment by nzchicken** [comment]
@Darel13712 you are more than welcome to write the tests yourself and submit a pr if you so desire. The build is passing because the thing builds. Build passing doesn't have to mean it works.

But more importantly, he doesn't technically owe us anything, that's what the license says. There is no warranty, etc. If I were to raise an issue on one or your repos, but you were on holiday, then what should I expect? One unpaid man owes us 0, and I'm just grateful he contributed to start off with.

**Comment by for7raid** [comment]
Guys, please, I never told you about warranty, I never told you drop all your business and quickly run to fix this issue.
Just ask yourself for what do you do public open source? Why you welcome only to respect, but not to criticism. If this project only hobby to feeling yourself - make it private, if you do it for helping to people, to get respect and acknowledgment - be welcome to any feedback. Please pay a simplest respect to users and talk to us, to everyone, who is confused like me, why bug is steel exists but the issue is closed without any commits, description, bug fix release.
Yes, nobody pay money for this job, but we pay respect, starts and likes to you, which can be change to real money, help you, when you go to find new good job. And it is "plus" to your karma, of course.
I'm the same developer as you all, and have a little open source project too, and fully understand what does it mean. Open source does not equal to free, open source does not equal to unresponsible. When Samuel reject my improvement request - it is fully ok to me, but close unresolved bug - very strange.
I'd like to make a PR to fix this, but master branch do not build on my windows machine, and again there is no any instruction how to run it. How can I help you, if you do not want to get my help?
I can test any CI build, but you do not provide any link, any ask for this.
How many downloads and active users did you have 3 months ago? And how many now? I think zero, because the app is unusable at all. Seems, the next good open source project is dying, I do afraid that never can use it again, and this is a reason why I cry here.
Now I go with WebFactoryLtd's suggest and uninstalled the app, with my sad.

R.I.P. GPMDP.

**Comment by DenisStetsenko** [comment]
Does anyone know how to get back Chromecast button?

`Mac OS Catalina 10.15.2 (19C57)`

**Comment by pete360** [comment]
> Windows 10.
> 
> Well, I got a little further but now have a new issue. I installed Revo Uninstaller and did a complete uninstall and wipe of GPMD. Did a fresh install and it bring me to sign in the page, after signing I get a message from Google:
> 
> > Couldn't sign you in
> > This browser or app may not be secure. Learn more
> > Try using a different browser. If you’re already using a supported browser, you can refresh your screen and try again to sign in.
> 
> ![image](https://user-images.githubusercontent.com/4324474/71420305-84724600-2642-11ea-89d0-fd83fab7f980.png)

Why was this marked off topic when I provided information as to what resolved the loading library issue. It was a clean uninstall and re-install. Not my fault it brought up a new issue.

Also, why is this thread closed considering the bug still exists?

**Comment by agc93** [comment]
@pete360 My best guess is its closed as the specific bug this was opened for (i.e. being stuck on the opening Loading Music Library) seems to have been fixed in current CI builds ([tool for anyone without a CI build](https://gpmdp.azurewebsites.net/) ).

I think that specific error is tracked in #3545 instead.

**Comment by hpeyerl** [comment]
Been involved in Open Source before it was called that.  The only thing I ever feel that I owe my users is a working build from an official source.  If I've broken the official build, then I either fix it or back out the change and re-release.  Open Source is sometimes a pain in the butt but it's not hard.

**Comment by Sonny812** [comment]
The following helped me on Fedora 31:
1. Remove GPMDP `$ sudo dnf remove google-play-music-desktop-player-4.6.1.x86_64.rpm`
2. Remove GPMDP config files
``` 
$ cd $HOME
$ rm -rf .config/Google\ Play\ Music\ Desktop\ Player/ 
```
3. Download the latest rpm x64 build from Circle CI
     * Go to https://circleci.com/vcs-authorize/
     * Sign in
     * Find here https://circleci.com/gh/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL- the latest successfully build with jobname "artifact_gather" and click it
     * Go to artifacts tab, find there the link to google-play-music-desktop-player-4.6.1.x86_64.rpm and copy it.
     * Download the build from the copied link `$ wget https://3838-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/redhat/google-play-music-desktop-player-4.6.1.x86_64.rpm` 
4. Install it using `$ sudo dnf install google-play-music-desktop-player-4.6.1.x86_64.rpm`

**Comment by MarshallOfSound** [comment]
As I said up in the original comment that closed this issue, this issue is fixed in master and will be released once we've got a the following things:

* The user agent issue in #3545 figured out
* A green build
* Me having spare time to actually organize and run a release

I'm going to lock this out for now to avoid notification spam, if you want CI builds but don't know how to get them it looks like an awesome community member (sorry not sure who originally built it) has made a handy tool for fetching CI builds.

https://gpmdp.azurewebsites.net/

**Comment by MarshallOfSound** [comment]
There are also `4.7.0` RC builds available over here --> https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3545#issuecomment-569379704

Check out that comment and follow the instructions there
